### PR TITLE
fixed openssl include for protocol fuzzing

### PR DESCRIPTION
**Pull request by ericwol-msft**
This is needed so that we can build the fuzzer under linux, without openssl and socketio enabled

### CODE REVIEW

File: adapters/platform_linux.c [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #include "azure_c_shared_utility/httpapiex.h"
 #ifdef USE_OPENSSL
 #include "azure_c_shared_utility/tlsio_openssl.h"
+#else
+const IO_INTERFACE_DESCRIPTION* tlsio_openssl_get_interface_description();
```
Comment (danewalton) [inline adapters/platform_linux.c:11]: wouldn't this need the definition which is in the header?

Comment (ericwol-msft) [reply adapters/platform_linux.c:11]: only if USE_OPENSSL is used, which will include it in tlsio_openssl.h"

Comment (danewalton) [reply adapters/platform_linux.c:11]: If this is used for fuzzing, should it be called something other than openssl? Or is it actually somehow using openssl?

Comment (ericwol-msft) [reply adapters/platform_linux.c:11]: note this is needed because of line 61

**Review by danewalton — APPROVED** [review]
looks good to me but maybe @ewertons weigh in here?

**Review by momuno — APPROVED** [review]

**Pull request merged**